api4/group.go

```diff
@@ -337,8 +337,9 @@ func patchGroupSyncable(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if !c.App.SessionHasPermissionTo(c.App.Session, model.PERMISSION_MANAGE_SYSTEM) {
-		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+	appErr := verifyLinkUnlinkPermission(c, syncableType, syncableID)
+	if appErr != nil {
+		c.Err = appErr
 		return
 	}
```

Comment:
Martin, are you going to integrate these changes into your PR then?

Reply:
@catalintomai Yes I have integrated it in. https://github.com/mattermost/mattermost-server/pull/13361/commits/9e6dedde15bfc6d01ac1a9518ff8a792a0357022. Thanks.